### PR TITLE
[Merged by Bors] - Remaining fn in Timer

### LINKED ISSUE #5963: Add a method on Timer to get the remaining time
## What problem does this solve or what need does it fill?

Getting the remaining time is a natural operation, and particularly useful when working with cooldowns or countdowns.

## What solution would you like?

Add a `remaining()` method to [`Timer`](https://docs.rs/bevy/latest/bevy/time/struct.Timer.html).

### PR DESCRIPTION
**Pull request by Sergi-Ferrez**
# Objective

Fixes #5963 

## Solution

Add remaining fn in Timer class, this function only minus total duration with elapsed time.

### CODE REVIEW

File: crates/bevy_time/src/timer.rs [hunk 1]
```diff
@@ -343,6 +343,21 @@ impl Timer {
         1.0 - self.percent()
     }
 
+    /// Returns the remainig time in seconds
```
Comment (rparrett) [inline crates/bevy_time/src/timer.rs:346]:
```suggestion
    /// Returns the remaining time in seconds
```

**Comment by rparrett** [comment]
For symmetry with `elapsed` and `elapsed_secs`, I think we should name this `remaining_secs` and add `remaining` that returns a `Duration`.

File: crates/bevy_time/src/timer.rs [hunk 2]
```diff
@@ -343,6 +343,21 @@ impl Timer {
...
+    /// # Examples
+    /// ```
+    /// # use bevy_time::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(2.0, false);
+    /// timer.tick(Duration::from_secs_f32(0.5));
+    /// assert_eq!(timer.remaining(), 1.5);
+    /// ```
+    #[inline]
+    pub fn remaining(&self) -> f32 {
```
Comment (alice-i-cecile) [inline crates/bevy_time/src/timer.rs:357]: This should return a `Duration` like `elapased` :) Users can easily convert that into an `f32` on their own.

File: crates/bevy_time/src/timer.rs [hunk 3]
```diff
@@ -343,6 +343,21 @@ impl Timer {
...
 
+    /// Returns the remainig time in seconds
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_time::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(2.0, false);
+    /// timer.tick(Duration::from_secs_f32(0.5));
+    /// assert_eq!(timer.remaining(), 1.5);
```
Comment (alice-i-cecile) [inline crates/bevy_time/src/timer.rs:354]: Float equality checking is a bad habit; we should have some error bars here.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]

**Comment by harudagondi** [comment]
> I purpose using [float-cmp](https://crates.io/crates/float-cmp) to properly compare floats in an error range.

We don't have to pull in another crate. I think [`f32::total_cmp`](https://doc.rust-lang.org/std/primitive.f32.html#method.total_cmp) should be fine for doctests.

**Comment by alice-i-cecile** [comment]
LGTM once the float equality check is addressed :)

**Comment by Sergi-Ferrez** [comment]
Other parts of timer implementation also use implicit float comparison. Maybe create another issue for fixing that?

**Comment by rparrett** [comment]
`total_cmp` doesn't really help with the sort of issue we're trying to avoid here.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=95d76d813f1d9c954a4e00323b129f52

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
(Oops, approved by accident; don't count it)

**Comment by mockersf** [comment]
In my opinion, the doc tests you added are not really needed and should be removed. But it's the current documentation style of `Timer`, so 🤷 

For me, having that many code examples that show "obvious" behaviours make docs less readable: https://docs.rs/bevy/latest/bevy/prelude/struct.Timer.html

Approving this PR as is, but I think an issue or a followup PR should be opened to remove most of the doc examples.

**Review by mockersf — APPROVED** [review]

File: crates/bevy_time/src/timer.rs [hunk 4]
```diff
@@ -343,6 +343,38 @@ impl Timer {
...
+    /// use std::cmp::Ordering;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(2.0, false);
+    /// timer.tick(Duration::from_secs_f32(0.5));
+    /// let result = timer.remaining_secs().total_cmp(&1.5);
+    /// assert_eq!(Ordering::Equal, result);
+    /// ```
+    #[inline]
+    pub fn remaining_secs(&self) -> f32 {
+        self.duration().as_secs_f32() - self.elapsed().as_secs_f32()
```
Comment (cart) [inline crates/bevy_time/src/timer.rs:360]: I think it probably makes sense to make this `self.remaining().as_secs_f32()`. That way we don't re-specify the "remaining" logic in multiple places, and we only do a single float conversion at the end (which might preserve precision in some cases).

Comment (Sergi-Ferrez) [reply crates/bevy_time/src/timer.rs:360]: I totally agree, going to change it

**Pull request closed without merging**